test/test_direct_serendipity.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import FIAT
+import finat
+import gem
+import numpy as np
+from finat.physically_mapped import PhysicalGeometry
+
+
+class MyMapping(PhysicalGeometry):
+    def __init__(self, cell, verts):
+        # cell is reference cell, verts is physical vertices
+        self.verts = np.asarray(verts)
+
+    def cell_size(self):
+        return gem.Literal(np.full(self.cell.num_vertices(), 1, dtype=np.int32))
+
+    def jacobian_at(self, point):
+        raise NotImplementedError
+
+    def reference_normals(self):
+        raise NotImplementedError
+
+    def physical_normals(self):
+        raise NotImplementedError
+
+    def physical_tangents(self):
+        raise NotImplementedError
+
+    def physical_edge_lengths(self):
+        raise NotImplementedError
+
+    def physical_points(self, ps, entity=None):
+        assert entity is None
+        prefs = ps.points
+        pvs = self.verts
+        pps = np.zeros(prefs.shape, dtype=float)
+        for i in range(pps.shape[0]):
+            pps[i, :] = (pvs[0, :] * (1-prefs[i, 0]) * (1-prefs[i, 1])
+                         + pvs[1, :] * (1-prefs[i, 0]) * prefs[i, 1]
+                         + pvs[2, :] * prefs[i, 0] * (1-prefs[i, 1])
+                         + pvs[3, :] * prefs[i, 0] * prefs[i, 1])
+        pps_gem = np.zeros(pps.shape, dtype=object)
+        for alpha in np.ndindex(pps.shape):
+            pps_gem[alpha] = gem.Literal(pps[alpha])
+        return gem.ListTensor(pps_gem)
+
+    def physical_vertices(self):
+        pvs = np.zeros(self.verts.shape, dtype=object)
+        for alpha in np.ndindex(pvs.shape):
+            pvs[alpha] = gem.Literal(self.verts[alpha])
+        return gem.ListTensor(pvs)
+
+
+def get_pts(cell, deg):
+    assert cell.shape == 11  # quadrilateral
```

Comment:
Please use the symbolic constant `QUADRILATERAL` instead of this magic number + comment.